src/span_ext.rs

```diff
@@ -0,0 +1,78 @@
+use crate::layer::{build_context, WithContext};
+use opentelemetry::api;
+
+/// `OpenTelemetrySpanExt` allows tracing spans to accept and return
+/// OpenTelemetry `SpanContext`s.
+pub trait OpenTelemetrySpanExt {
+    /// Associates `self` with a given `OpenTelemetry` trace, using
+    /// the provided parent context.
+    ///
+    /// ```rust
+    /// use opentelemetry::api::{self, HttpTextFormat};
+    /// use tracing_opentelemetry::OpenTelemetrySpanExt;
+    /// use std::collections::HashMap;
+    ///
+    /// // Example carrier, could be a framework header map that impls `api::Carrier`.
+    /// let mut carrier = HashMap::new();
+    ///
+    /// // Propagator can be swapped with trace context propagator binary propagator, etc.
+    /// let propagator = api::B3Propagator::new(true);
+    ///
+    /// // Extract otel parent span context via the chosen propagator
+    /// let parent_context = propagator.extract(&carrier);
+    ///
+    /// // Generate a tracing span as usual
+    /// let app_root = tracing::span!(tracing::Level::INFO, "app_start");
+    ///
+    /// // Assign parent trace from external context
+    /// app_root.set_opentelemetry_parent(parent_context);
+    /// ```
+    fn set_opentelemetry_parent(&self, span_context: api::SpanContext);
```

Comment:
yeah I originally had `set_parent` / `context` but was worried about the clashing. How likely do you think that folks will import multiple span extensions at this point where it might start to become an issue? Would definitely prefer to have the shorter names.